src/components/JourneyPicker/JourneyPicker.tsx

```diff
@@ -95,44 +96,55 @@ const JourneyPicker = (props: Props) => {
         day,
         fromStationId,
         onSelectDay,
-        onSelectJourney,
+        updateJourneyParams,
         onSelectTimeOfDay,
         stationsById,
         stationsByLine,
         time,
         timeRange,
         toStationId,
         disabled,
-        reverse,
+        navigationKind,
     } = props;
 
     const [timeOfDay, setTimeOfDay] = useState(() =>
         typeof time === "number" ? getTimeOfDayOptionForTime(time) : timeOfDayPickerOptions[0]
     );
 
+    const [selectedNavigationKind, setSelectedNavigationKind] = useState(
```

Comment:
I would consider replacing this with:

```ts
const selectedNavigationKind = useMemo(() => {
    return navigationKindOptions.find(o => o.id === navigationKind) || navigationKindOptions[0];
}, [navigationKind]);
```

This lets you drop the following:

- A `useState()` hook whose value needs to be carefully synced with a prop
- A new callback, `chooseDepartureOption()`, that exists to do the syncing
- A new `useEffect()` hook that forces React to render twice whenever this component changes.

In the general case, I do see a downside to using `array.find()` on every render. But that's not such a big deal when the array is very small, and we can avoid the problem completely with `useMemo()`.